Error while parsing WhatsApp contacts

This is a rare issue reported by another user.
WhatsApp contacts were missing from a processed UFDR.
Looking at the processing log, there was a AIOOB exception:
```text
org.apache.tika.exception.TikaException: WAExtractorException Exception
	at iped.parsers.whatsapp.WhatsAppParser.parseWhatsAppContacts(WhatsAppParser.java:1186)
	at iped.parsers.whatsapp.WhatsAppParser.parse(WhatsAppParser.java:254)
	at org.apache.tika.parser.CompositeParser.parse(CompositeParser.java:298)
	at iped.parsers.standard.StandardParser.parse(StandardParser.java:245)
	at iped.engine.io.ParsingReader$BackgroundParsing.run(ParsingReader.java:247)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
	at iped.parsers.whatsapp.WAContact.<init>(WAContact.java:31)
	at iped.parsers.whatsapp.WAContactsDirectory.getContact(WAContactsDirectory.java:13)
	at iped.parsers.whatsapp.WAContactsExtractorAndroid.extractContactList(WAContactsExtractorAndroid.java:67)
	at iped.parsers.whatsapp.WhatsAppParser.parseWhatsAppContacts(WhatsAppParser.java:1143)
	... 9 more
2023-09-14 18:13:28 [WARN] [parsers.standard.StandardParser] ParsingThread-48 Parsing exception: XXXXXXXXX/apps/com.whatsapp/db/wa.db (196608 bytes) java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
```

Digging into the problem, I found that one recovered record from the "wa.db" contained a string terminated with "@" as the JID (something like "5511987654321@").

https://github.com/sepinf-inc/IPED/blob/ef18f22c2118f533ed4bdf7502d50fafbe03855b/iped-parsers/iped-parsers-impl/src/main/java/iped/parsers/whatsapp/WAContact.java#L28-L31

The exception is throw because `id.contains("@")` is true but `"5511987654321@".split("@")` returns a single element.

--- a/iped-parsers/iped-parsers-impl/src/main/java/iped/parsers/whatsapp/WAContact.java
+++ b/iped-parsers/iped-parsers-impl/src/main/java/iped/parsers/whatsapp/WAContact.java
@@ -27,8 +27,8 @@ public class WAContact {
     public WAContact(String id) {
         if (id != null) {
             String[] idSplit = id.split("@", 2);
-            this.id = idSplit[0];
-            this.suffix = idSplit.length > 1 ? idSplit[1] : "";
+            this.id = idSplit[0].trim();
+            this.suffix = idSplit.length > 1 ? idSplit[1].trim() : "";
         } else {
             this.id = this.suffix = "";
         }

--- a/iped-parsers/iped-parsers-impl/src/main/java/iped/parsers/whatsapp/WAContact.java
+++ b/iped-parsers/iped-parsers-impl/src/main/java/iped/parsers/whatsapp/WAContact.java
@@ -25,13 +25,12 @@ public class WAContact {
     private boolean deleted = false;
 
     public WAContact(String id) {
-        if (id != null && id.contains("@")) { //$NON-NLS-1$
-            String[] id_split = id.split("@"); //$NON-NLS-1$
-            this.id = id_split[0];
-            this.suffix = id_split[1];
+        if (id != null) {
+            String[] idSplit = id.split("@", 2);
+            this.id = idSplit[0];
+            this.suffix = idSplit.length > 1 ? idSplit[1] : "";
         } else {
-            this.id = id;
-            this.suffix = ""; //$NON-NLS-1$
+            this.id = this.suffix = "";
         }
     }
 

